File: src/main/java/utils/codec/Ascii.java
package utils.codec;

import java.nio.charset.StandardCharsets;

public class Ascii {
    public static byte[] fromAscii(String ascii) {
        return ascii.getBytes();
    }

    public static String toAscii(byte[] bytes) {
        return new String(bytes, StandardCharsets.US_ASCII);
    }
}

File: src/main/kotlin/utils/codec/Ascii.kt
@file:OptIn(ExperimentalUnsignedTypes::class)

package utils.codec

import java.nio.charset.StandardCharsets

@JvmInline
value class Ascii(val string: String)

fun Ascii.toBytes(): UByteArray {
    val ascii = this
    return ascii.string.toByteArray().toUByteArray()
}

fun UByteArray.toAscii(): Ascii {
    return Ascii(String(this.toByteArray(), StandardCharsets.US_ASCII))
}
